src/TransparentValueObjects.Augments/IHasSystemTextJson.cs

```diff
@@ -0,0 +1,8 @@
+namespace TransparentValueObjects.Augments;
+
+public interface IHasSystemTextJson<out TValueObject, out TValue>
+    where TValueObject : IValueObject<TValue>
+    where TValue : notnull
+{
+
+}
```

Comment:
The interface should contain the converter type:

```csharp
public interface IHasJsonConverter
{
    public static abstract Type JsonConverterType { get; }
}
```

If the user doesn't implement this interface, we'll auto-generate a converter and implement this interface. We can use the `JsonConverterType` field in the `JsonConverterAttribute`:

```csharp
[ValueObject<string>]
[JsonConverter(typeof(MyJsonConverter))]
readonly partial struct MyValueObject
```

If the user implements the interface, we'd just have to parse the expression and put it into the attribute which will be auto-generated.

Reply:
That was a good idea. I'm not sure if `IHasSystemTextJson` is needed now at all, so I've removed it. Should be easy to add back
I also renamed `IHasJsonConverter` to `IHasSystemTextJsonConverter` because we could potentially support NewtonsoftJson